services/api/config/arvados_config.rb

```diff
@@ -270,25 +271,22 @@ def arrayToHash cfg, k, v
 if $arvados_config["PostgreSQL"]["Connection"]["port"] != 0
   dbhost += ":#{$arvados_config["PostgreSQL"]["Connection"]["port"]}"
 end
- 
-if $arvados_config["PostgreSQL"]["Connection"]["user"].include? "@"
-  user_string = $arvados_config["PostgreSQL"]["Connection"]["user"]
-  $arvados_config["PostgreSQL"]["Connection"]["user"] = user_string.gsub("@", "%40")
-end
 
 #
 # If DATABASE_URL is set, then ActiveRecord won't error out if database.yml doesn't exist.
 #
 # For config migration, we've previously populated the PostgreSQL
 # section of the config from database.yml
 #
-ENV["DATABASE_URL"] = "postgresql://#{$arvados_config["PostgreSQL"]["Connection"]["user"]}:"+
+database_url = CGI.escape("postgresql://#{$arvados_config["PostgreSQL"]["Connection"]["user"]}:"+
```

Comment:
I don't think that is going to work, that would escape the `@` and `:` characters that actually need to be there.  Should go inside the string interpolations:

```
database_url = "postgresql://#{CGI.escape $arvados_config["PostgreSQL"]["Connection"]["user"]}:"+
```

Reply:
That would make sense, sorry this is my first foray into ruby.

Reply:
No worries, we are very happy for the contribution!